ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -67,6 +68,8 @@ const mapStateToProps = (state, props) => {
   const transaction = R.find(({ id }) => id === transactionId)(selectedAddressTxList)
   const transactionStatus = transaction ? transaction.status : ''
 
+  const unapprovedTxCount = valuesFor(unapprovedTxs).length
```

Comment:
Check the `unconfirmedTransactionsCountSelector` of `metamask-extension/ui/app/selectors/confirm-transaction.js`

That counts messages as well, so you can't use it in this case.

I bring it up to highlight the way it gets the count for unapprovedTxs (see `filteredUnapprovedTxIds`). The filtering there is done to ensure that only txs from the current network are counted. I believe this is necessary because, currently, shapeshift txs will appear on all network. @alextsg can confirm

If so, you'll need to do the same here.

Reply:
This is a good point, but isn't rejecting messages something we want to include in Reject All? Is that not the behavior of the old UI?

Reply:
> The filtering there is done to ensure that only txs from the current network are counted. I believe this is necessary because, currently, shapeshift txs will appear on all network.

Agree that this filtering is needed, I'll add that

> This is a good point, but isn't rejecting messages something we want to include in Reject All? Is that not the behavior of the old UI?

The old UI does reject everything in `unapprovedTxs` but I think it's confusing to translate that 1:1 to the new UI because there's a lot more separation of confirm screens and I think it would be confusing the user.